plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolConnectorTest.java

```diff
@@ -0,0 +1,472 @@
+                .hasMessageContaining("Query not supported: ResultSetMetaData not available for query: CREATE TABLE ");
+        assertFalse(getQueryRunner().tableExists(getSession(), "numbers"));
+    }
+
+    @Override
+    @Test
+    public void testDateYearOfEraPredicate()
+    {
+        // Override because Exasol does not support negative dates
+        assertQuery("SELECT orderdate FROM orders WHERE orderdate = DATE '1997-09-14'", "VALUES DATE '1997-09-14'");
+    }
+
+    @Override
+    @Test
+    public void testInformationSchemaFiltering()
+    {
+        // Exasol reports bigint columns as decimal(19,0)
+        assertQuery(
+                "SELECT table_name FROM information_schema.tables WHERE table_name = 'orders' LIMIT 1",
+                "SELECT 'orders' table_name");
+        assertQuery(
+                "SELECT table_name FROM information_schema.columns WHERE data_type = 'decimal(19,0)' AND table_name = 'customer' AND column_name = 'custkey' LIMIT 1",
+                "SELECT 'customer' table_name");
+    }
+
+    private void predicatePushdownTest(String exasolType, String exasolLiteral, String operator, String filterLiteral)
+    {
+        String tableName = "test_pdown_" + exasolType.replaceAll("[^a-zA-Z0-9]", "");
+        try (TestTable table = new TestTable(onRemoteDatabase(), TEST_SCHEMA + "." + tableName, format("(c %s)", exasolType))) {
+            onRemoteDatabase().execute(format("INSERT INTO %s VALUES (%s)", table.getName(), exasolLiteral));
+
+            assertThat(query(format("SELECT * FROM %s WHERE c %s %s", table.getName(), operator, filterLiteral)))
+                    .isFullyPushedDown();
+        }
+    }
+
+    @Override
+    @Test
+    public void testCharTrailingSpace()
+    {
+        abort("Exasol pads CHAR(n) with spaces");
+    }
+
+    @Override
+    @Test
+    public void testNativeQuerySimple()
+    {
+        // Override because Exasol requires named columns
+        assertQuery("SELECT * FROM TABLE(system.query(query => 'SELECT 1 AS res'))", "VALUES 1");
+    }
+
+    @Test
+    public void testNativeMultipleInClauses()
+    {
+        String longInClauses = range(0, 10)
+                .mapToObj(value -> getLongInClause(value * 1_000, 1_000))
+                .collect(joining(" OR "));
+        onRemoteDatabase().execute(format("SELECT count(*) FROM %s.orders WHERE %s", TEST_SCHEMA, longInClauses));
+    }
+
+    private String getLongInClause(int start, int length)
+    {
+        String longValues = range(start, start + length)
+                .mapToObj(Integer::toString)
+                .collect(joining(", "));
+        return "orderkey IN (" + longValues + ")";
+    }
+
+    @Override
+    protected SqlExecutor onRemoteDatabase()
+    {
+        return exasolServer::execute;
+    }
+
+
+    @Override
+    @Test
+    public void testInsertInPresenceOfNotSupportedColumn() {
+        abort("Exasol does not support insert");
```

Comment:
Please use `assertThatThrownBy`. Otherwise, we will forget to update this method when supporting INSERT statement.